src/Domain/CoordinateMaps/CoordinateMap.hpp

```diff
@@ -358,397 +358,46 @@ class CoordinateMap
-
-  return tnsr::I<T, dim, TargetFrame>(std::move(mapped_point));
-}
-
-template <typename SourceFrame, typename TargetFrame, typename... Maps>
-template <typename T, size_t... Is>
-constexpr SPECTRE_ALWAYS_INLINE boost::optional<tnsr::I<
-    T, CoordinateMap<SourceFrame, TargetFrame, Maps...>::dim, SourceFrame>>
-CoordinateMap<SourceFrame, TargetFrame, Maps...>::inverse_impl(
-    tnsr::I<T, dim, TargetFrame>&& target_point, const double time,
-    const std::unordered_map<
-        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
-        functions_of_time,
-    std::index_sequence<Is...> /*meta*/) const noexcept {
-  boost::optional<std::array<T, dim>> mapped_point(
-      make_array<T, dim>(std::move(target_point)));
-
-  (void)std::initializer_list<char>{make_overloader(
-      [](const auto& the_map, boost::optional<std::array<T, dim>>& point,
-         const double /*t*/,
-         const std::unordered_map<
-             std::string,
-             std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
-         /*funcs_of_time*/,
-         const std::false_type /*is_time_independent*/) noexcept {
-        if (point) {
-          if (LIKELY(not the_map.is_identity())) {
-            point = the_map.inverse(point.get());
-          }
-        }
-        return '0';
-      },
-      [](const auto& the_map, boost::optional<std::array<T, dim>>& point,
-         const double t,
-         const std::unordered_map<
-             std::string,
-             std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
-             funcs_of_time,
-         const std::true_type /*is_time_dependent*/) noexcept {
-        if (point) {
-          point = the_map.inverse(point.get(), t, funcs_of_time);
-        }
-        return '0';
-        // this is the inverse function, so the iterator sequence below is
-        // reversed
-      })(std::get<sizeof...(Maps) - 1 - Is>(maps_), mapped_point, time,
-         functions_of_time,
-         CoordinateMap_detail::is_map_time_dependent_t<decltype(
-             std::get<sizeof...(Maps) - 1 - Is>(maps_))>{})...};
-
-  return mapped_point
-             ? tnsr::I<T, dim, SourceFrame>(std::move(mapped_point.get()))
-             : boost::optional<tnsr::I<T, dim, SourceFrame>>{};
-}
-
-// define type-trait to check for time-dependent jacobian
-namespace CoordinateMap_detail {
-CREATE_IS_CALLABLE(jacobian)
-template <typename Map, typename T>
-using is_jacobian_time_dependent_t =
-    CoordinateMap_detail::is_jacobian_callable_t<
-        Map, std::array<std::decay_t<T>, std::decay_t<Map>::dim>, double,
-        std::unordered_map<
-            std::string,
-            std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>>;
-}  // namespace CoordinateMap_detail
-
-template <typename SourceFrame, typename TargetFrame, typename... Maps>
-template <typename T>
-constexpr SPECTRE_ALWAYS_INLINE auto
-CoordinateMap<SourceFrame, TargetFrame, Maps...>::inv_jacobian_impl(
```

Comment:
Ah, I see what you mean. Since these templates are called by virtual functions and the virtual functions are instantiated whenever the class is instantiated, so will the `_impl` functions. This mean that they `tpp` only needs to be included in the `cpp` file and the instantiation of these functions will be done because we need the constructor and therefore also the whole class. It's sort of a cascade. Does that make sense?

Reply:
Gotcha, thanks! :+1: